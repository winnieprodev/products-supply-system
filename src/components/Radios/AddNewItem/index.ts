export { default as AddNewItem } from "./AddNewItem";
