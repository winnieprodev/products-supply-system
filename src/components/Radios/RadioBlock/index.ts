export { default as RadioBlock } from "./RadioBlock";
