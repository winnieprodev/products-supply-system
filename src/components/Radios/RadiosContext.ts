import React from "react";

export interface RadiosContextProps {
  selected: string;
  setSelected: (value: string) => void;
}

export const RadiosContext = React.createContext<RadiosContextProps>({
  selected: "0",
  setSelected: () => {},
});
