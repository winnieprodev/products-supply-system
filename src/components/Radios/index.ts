export { default as Radios } from "./Radios";
