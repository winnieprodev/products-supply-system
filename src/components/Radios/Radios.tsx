import { AddNewItem } from "./AddNewItem";
import { RadioBlock } from "./RadioBlock";
import * as Styled from "./radios.styled";
import { RadiosContext, RadiosContextProps } from "./RadiosContext";

type OwnProps = RadiosContextProps;

const Radios = ({
  selected,
  setSelected,
  children,
}: React.PropsWithChildren<OwnProps>) => (
  <RadiosContext.Provider value={{ selected, setSelected }}>
    <Styled.Wrapper>{children}</Styled.Wrapper>
  </RadiosContext.Provider>
);

Radios.RadioBlock = RadioBlock;
Radios.AddNewItem = AddNewItem;

export default Radios;
