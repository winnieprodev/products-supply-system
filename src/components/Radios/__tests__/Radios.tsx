import { render } from "@testing-library/react";

import Radios from "../Radios";

describe("Radios", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot", () => {
      const { container } = render(
        <Radios selected="a" setSelected={jest.fn()}>
          test
        </Radios>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
