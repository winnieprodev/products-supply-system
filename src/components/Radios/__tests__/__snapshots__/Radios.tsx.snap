// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radios Snapshot tests it matches the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 16px;
}

<div
  class="c0"
>
  test
</div>
`;
